core-plugins/src/main/java/io/cdap/plugin/batch/source/ExcelInputReader.java

```diff
@@ -143,14 +153,16 @@ private void init() {
       for (String schema : schemaList) {
         String[] columns = schema.split(":");
         if (CollectionUtils.isNotEmpty(inputColumns) && !inputColumns.contains(columns[0])) {
-          throw new IllegalArgumentException("Column name: " + columns[0] + " in 'Field Name Schema Type Mapping'" +
-                                               " does not match the columns in the 'Column To Be Extracted' input " +
-                                               "text box. It has to be one of the columns in " +
-                                               "'Column To Be Extracted' input text box.");
+          collector.addFailure(
+              String.format("Column: %s in 'Field Name Schema Type Mapping' "
+                  + "must be included in 'Column To Be Extracted'", columns[0]), null)
+              .withConfigElement(OUTPUT_SCHEMA, schema);
+        } else {
+          outputSchemaMapping.put(columns[0], columns[1]);
         }
-        outputSchemaMapping.put(columns[0], columns[1]);
       }
     }
+    collector.getOrThrowException();
```

Comment:
whether to call this or not should be decided by the caller and not placed here at the end of the method

Reply:
Fixed